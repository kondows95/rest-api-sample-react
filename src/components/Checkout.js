import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Grid, Box,TextField } from '@material-ui/core';
import CheckoutShippingInformation from '../containers/CheckoutShippingInformation';
import CheckoutOrderReviewCart from '../containers/CheckoutOrderReviewCart';
import CheckoutPayment from '../containers/CheckoutPayment';
const useStyles = makeStyles(theme => ({
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));

export default function AddressForm() {
  const classes = useStyles()
  return (
    <form className={classes.form} noValidate>
      <CheckoutShippingInformation />
      <CheckoutOrderReviewCart />
      <CheckoutPayment />
    </form>
  );
}