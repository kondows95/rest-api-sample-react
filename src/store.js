import rootReducer from './modules/index';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
export default createStore(rootReducer, applyMiddleware(thunk));