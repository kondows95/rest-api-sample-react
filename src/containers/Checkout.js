import { connect } from 'react-redux'
import Checkout from '../components/Checkout'
import { setRequestParams, postOrder } from '../modules/order.ts'

export default connect(
  (state) => ({
    requestParams: state.order.requestParams,
    postResultObj: state.order.postResultObj
  }),
  (dispatch) => ({
    setRequestParams: (addressForm) =>  dispatch(setRequestParams(addressForm)),
    postOrder: () =>  dispatch(postOrder())
  })
)(Checkout)