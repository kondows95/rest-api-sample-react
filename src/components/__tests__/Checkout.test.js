import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import Checkout from '../Checkout';
import { Provider } from 'react-redux'
import store from '../../store';
import { BrowserRouter as Router } from 'react-router-dom';
let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})

const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;
const requestParams = {
  first_name: "",
  last_name: "",
  address1: "",
  address2: "",
  country: "",
  state: "",
  city: ""
}
const setRequestParams = jest.fn((form) => {
  console.log("setRequestParams  ",form)
  let setRequest ={...form};
});
describe("CartTable component", () => {
  it('matches the snapshot', () => {
    const CheckoutSnapshot = renderer.create(<Provider store={store}>
      <Router>
        <Checkout requestParams={requestParams} />
      </Router>
    </Provider>).toJSON();
    expect(CheckoutSnapshot).toMatchSnapshot();
  })
})


describe("testing Checkout ", () => {
  it('testing textfield handelchange and click comfirm button', () => {
    act(() => {
      ReactDOM.render((<Provider store={store}>
        <Router>
          <Checkout requestParams={requestParams} setRequestParams={setRequestParams} />
        </Router>
      </Provider>), container);
    });
    const inputArr = document.querySelectorAll('input');
    act(() => {
      valueSetter.call(inputArr[0], "wint");
      inputArr[0].dispatchEvent(new Event('change', { bubbles: true }));
    });
    act(() => {
      valueSetter.call(inputArr[1], "shwe yee");
      inputArr[1].dispatchEvent(new Event('change', { bubbles: true }));
    });
    act(() => {
      valueSetter.call(inputArr[2], "ye mon taung");
      inputArr[2].dispatchEvent(new Event('change', { bubbles: true }));
    });
    act(() => {
      valueSetter.call(inputArr[3], "mandalay");
      inputArr[3].dispatchEvent(new Event('change', { bubbles: true }));
    });
    act(() => {
      valueSetter.call(inputArr[4], "myanmar");
      inputArr[4].dispatchEvent(new Event('change', { bubbles: true }));
    });
    act(() => {
      valueSetter.call(inputArr[5], "mandalay");
      inputArr[5].dispatchEvent(new Event('change', { bubbles: true }));
    });
    act(() => {
      valueSetter.call(inputArr[6], "mandalay");
      inputArr[6].dispatchEvent(new Event('change', { bubbles: true }));
    });
    expect(inputArr[0].value).toBe("wint");
    expect(inputArr[1].value).toBe("shwe yee");
    expect(inputArr[2].value).toBe("ye mon taung");
    expect(inputArr[3].value).toBe("mandalay");
    expect(inputArr[4].value).toBe("myanmar");
    expect(inputArr[5].value).toBe("mandalay");
    expect(inputArr[6].value).toBe("mandalay");
    const buttonArray = document.querySelectorAll('button');
    expect(buttonArray[2].textContent.toUpperCase()).toBe("PROCEED TO COMFIRM")
    act(() => {
      buttonArray[2].dispatchEvent(new Event('click', { bubbles: true }))
    });
    expect(setRequestParams).toHaveBeenCalled();
  });
 
})

