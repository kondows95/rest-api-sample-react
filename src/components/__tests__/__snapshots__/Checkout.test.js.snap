// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTable component matches the snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-4"
>
  <form
    className="makeStyles-form-1"
    onSubmit={[Function]}
  >
    <div
      className="MuiBox-root MuiBox-root-5"
    >
      1.Shipping Information
    </div>
    <div
      className="MuiBox-root MuiBox-root-6"
    >
      IF SHIPPING TO A WORK ADDRESS, PLEASE INCLUDE THE COMPANY NAME
    </div>
    <div
      className="MuiBox-root MuiBox-root-7"
    >
      YOUR INFORMATION
    </div>
    <div
      className="MuiBox-root MuiBox-root-8"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
        >
          <div
            className="MuiBox-root MuiBox-root-112"
          >
            First Name *
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            style={
              Object {
                "paddingRight": 8,
              }
            }
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-150 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-151"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="first_name"
                name="first_name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
        >
          <div
            className="MuiBox-root MuiBox-root-152"
          >
            Last Name
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            style={
              Object {
                "paddingRight": 8,
              }
            }
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-150 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-151"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="last_name"
                name="last_name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
        >
          <div
            className="MuiBox-root MuiBox-root-153"
          >
            Address1 *
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            style={
              Object {
                "paddingRight": 8,
              }
            }
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-150 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-151"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="address1"
                name="address1"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
        >
          <div
            className="MuiBox-root MuiBox-root-154"
          >
            Address2
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            style={
              Object {
                "paddingRight": 8,
              }
            }
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-150 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-151"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="address2"
                name="address2"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
        >
          <div
            className="MuiBox-root MuiBox-root-155"
          >
            Country *
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            style={
              Object {
                "paddingRight": 8,
              }
            }
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-150 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-151"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="country"
                name="country"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
        >
          <div
            className="MuiBox-root MuiBox-root-156"
          >
            State
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            style={
              Object {
                "paddingRight": 8,
              }
            }
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-150 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-151"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="state"
                name="state"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
        >
          <div
            className="MuiBox-root MuiBox-root-157"
          >
            City *
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            style={
              Object {
                "paddingRight": 8,
              }
            }
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-150 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-151"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="city"
                name="city"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-158"
    >
      2.Order Review
    </div>
    <div
      className="MuiBox-root MuiBox-root-159"
    >
      SELECT SHIPPING METHOD
    </div>
    <div
      className="MuiBox-root MuiBox-root-160"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-166 MuiRadio-root"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <input
            aria-label="E"
            className="PrivateSwitchBase-input-169"
            name="radio-button-demo"
            onChange={[Function]}
            type="radio"
          />
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </span>
      FedEx-3-4 business day(s)
    </div>
    <div
      className="MuiBox-root MuiBox-root-191"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiBox-root MuiBox-root-223"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4"
          >
            <div
              className="MuiBox-root MuiBox-root-224"
            >
               
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4"
          >
            <div
              className="MuiBox-root MuiBox-root-225"
            >
               SUBTOTAL 
            </div>
            <div
              className="MuiBox-root MuiBox-root-226"
            >
               SHIPPING COSTS 
            </div>
            <div
              className="MuiBox-root MuiBox-root-227"
            >
               ADDITIONAL SHIPPING COSTS
            </div>
            <div
              className="MuiBox-root MuiBox-root-228"
            >
              SALES TAX
            </div>
            <div
              className="MuiBox-root MuiBox-root-229"
            >
              TOTAL COSTS(4 ARTICLSE)
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4"
          >
            <div
              className="MuiBox-root MuiBox-root-230"
            >
               
              0
               MMK 
            </div>
            <div
              className="MuiBox-root MuiBox-root-231"
            >
               0.0 MMK 
            </div>
            <div
              className="MuiBox-root MuiBox-root-232"
            >
               0.0 MMK 
            </div>
            <div
              className="MuiBox-root MuiBox-root-233"
            >
               0.0 MMK 
            </div>
            <div
              className="MuiBox-root MuiBox-root-234"
            >
               
              0
               MMK
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-239"
      dispaly="column"
    >
      <div
        className="MuiBox-root MuiBox-root-240"
      >
        2.Payment
      </div>
      <div
        className="MuiBox-root MuiBox-root-241"
      >
        SELECT PAYMENT METHOD
      </div>
      <div
        className="MuiBox-root MuiBox-root-242"
      >
        <div
          className="MuiBox-root MuiBox-root-243"
        >
          <div
            bgcolor="white"
            className="MuiGrid-root"
          >
            <div
              className="MuiTabs-root"
            >
              <div
                className="MuiTabs-scroller MuiTabs-fixed"
                onScroll={[Function]}
                role="tablist"
                style={
                  Object {
                    "marginBottom": null,
                    "overflow": "hidden",
                  }
                }
              >
                <div
                  className="MuiTabs-flexContainer"
                >
                  <button
                    aria-selected={true}
                    className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit makeStyles-default_tabStyle-236 Mui-selected MuiTab-labelIcon"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="tab"
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiTab-wrapper"
                    >
                      <div
                        className="MuiCardMedia-root"
                        style={
                          Object {
                            "backgroundImage": "url(\\"https://www.paisabazaar.com/wp-content/uploads/2018/06/premium-credit-cards.jpg\\")",
                            "height": 50,
                            "width": 100,
                          }
                        }
                        title="credit"
                      />
                      CREDIT CARD
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-selected={false}
                    className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit MuiTab-labelIcon"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="tab"
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiTab-wrapper"
                    >
                      <div
                        className="MuiCardMedia-root"
                        style={
                          Object {
                            "backgroundImage": "url(\\"https://cdn0.tnwcdn.com/wp-content/blogs.dir/1/files/2014/04/Screen-Shot-2014-04-30-at-18.18.55-730x321.png\\")",
                            "height": 50,
                            "width": 100,
                          }
                        }
                        title="paypal"
                      />
                      PAY PAL
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <span
                  className="PrivateTabIndicator-root-265 PrivateTabIndicator-colorSecondary-267 MuiTabs-indicator makeStyles-indicator-238"
                  style={
                    Object {
                      "left": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root"
        >
          <div
            className="MuiTypography-root MuiTypography-body1"
            style={
              Object {
                "padding": 24,
              }
            }
          >
            Please ensure the billing address matches the address held by the card user
            <div
              className="MuiGrid-root MuiGrid-container"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
              >
                <div
                  className="MuiBox-root MuiBox-root-298"
                >
                  Name on card *
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
              >
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
                  style={
                    Object {
                      "paddingRight": 8,
                    }
                  }
                >
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                    onClick={[Function]}
                  >
                    <fieldset
                      aria-hidden={true}
                      className="PrivateNotchedOutline-root-150 MuiOutlinedInput-notchedOutline"
                      style={
                        Object {
                          "paddingLeft": 8,
                        }
                      }
                    >
                      <legend
                        className="PrivateNotchedOutline-legend-151"
                        style={
                          Object {
                            "width": 0.01,
                          }
                        }
                      >
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "&#8203;",
                            }
                          }
                        />
                      </legend>
                    </fieldset>
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                      disabled={false}
                      id="card_name"
                      name="card_name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
              >
                <div
                  className="MuiBox-root MuiBox-root-299"
                >
                  Credit card number *
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
              >
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
                  style={
                    Object {
                      "paddingRight": 8,
                    }
                  }
                >
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                    onClick={[Function]}
                  >
                    <fieldset
                      aria-hidden={true}
                      className="PrivateNotchedOutline-root-150 MuiOutlinedInput-notchedOutline"
                      style={
                        Object {
                          "paddingLeft": 8,
                        }
                      }
                    >
                      <legend
                        className="PrivateNotchedOutline-legend-151"
                        style={
                          Object {
                            "width": 0.01,
                          }
                        }
                      >
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "&#8203;",
                            }
                          }
                        />
                      </legend>
                    </fieldset>
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                      disabled={false}
                      id="card_number"
                      name="card_number"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                    />
                  </div>
                </div>
                <small>
                  Example:4111222211112222
                </small>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
              >
                <div
                  className="MuiBox-root MuiBox-root-300"
                >
                  Expiry Date *
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
              >
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
                  style={
                    Object {
                      "paddingRight": 8,
                    }
                  }
                >
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                    onClick={[Function]}
                  >
                    <fieldset
                      aria-hidden={true}
                      className="PrivateNotchedOutline-root-150 MuiOutlinedInput-notchedOutline"
                      style={
                        Object {
                          "paddingLeft": 8,
                        }
                      }
                    >
                      <legend
                        className="PrivateNotchedOutline-legend-151"
                        style={
                          Object {
                            "width": 0.01,
                          }
                        }
                      >
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "&#8203;",
                            }
                          }
                        />
                      </legend>
                    </fieldset>
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                      disabled={false}
                      id="card_expirydate"
                      name="card_expirydate"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
              >
                <div
                  className="MuiBox-root MuiBox-root-301"
                >
                  Security *
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
              >
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
                  style={
                    Object {
                      "paddingRight": 8,
                    }
                  }
                >
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                    onClick={[Function]}
                  >
                    <fieldset
                      aria-hidden={true}
                      className="PrivateNotchedOutline-root-150 MuiOutlinedInput-notchedOutline"
                      style={
                        Object {
                          "paddingLeft": 8,
                        }
                      }
                    >
                      <legend
                        className="PrivateNotchedOutline-legend-151"
                        style={
                          Object {
                            "width": 0.01,
                          }
                        }
                      >
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "&#8203;",
                            }
                          }
                        />
                      </legend>
                    </fieldset>
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                      disabled={false}
                      id="card_security"
                      name="card_security"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                    />
                  </div>
                </div>
                <small>
                  (What is this?)
                </small>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-302"
    >
      By clicking 'PROCEED TO PAYMENT' you agree to our Terms and Conditions and you have reviewed your and personal details.
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          PROCEED TO COMFIRM
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
